plugins/main/public/components/overview/vulnerabilities/searchbar/use-seach-bar-configuration.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { useEffect, useState } from 'react'
+import { SearchBarProps, FilterManager, TimeRange, Query } from '../../../../../../../src/plugins/data/public'
+import { Filter, IIndexPattern, IndexPatternsContract } from '../../../../../../../src/plugins/data/public';
+import { getPlugins, getDataPlugin } from '../../../../kibana-services';
+
+import { useFilterManager, useQueryManager, useTimeFilter } from '../../../common/hooks';
+import { AUTHORIZED_AGENTS } from '../../../../../common/constants';
+
+// Input - types
+type tUseSearchBarCustomInputs = {
+    defaultIndexPatternID?: IIndexPattern['id'];
+    onFiltersUpdated?: (filters: Filter[]) => void;
+    onQuerySubmitted?: (payload: { dateRange: TimeRange; query?: Query }, isUpdate?: boolean) => void;
+}
+type tUseSearchBarProps =  Partial<SearchBarProps> & tUseSearchBarCustomInputs;
+
+// Output types
+type tUserSearchBarResponse = {
+    searchBarProps: Partial<SearchBarProps>;
+}
+
+/**
+ * Hook used to compose the searchbar configuration props
+ * @param props 
+ * @returns 
+ */
+const useSearchBarConfiguration = (props?: tUseSearchBarProps): tUserSearchBarResponse => {
+    // dependencies
+    const filterManager = useFilterManager().filterManager as FilterManager;
+    const [query, setQuery] = props?.query ? useState(props?.query) : useQueryManager();
+    const { timeFilter, timeHistory, setTimeFilter } = useTimeFilter();
+    // states
+    const [isLoading, setIsLoading] = useState(false);
+    const [indexPatternSelected, setIndexPatternSelected] = useState<IIndexPattern>();
+    const [error, setError] = useState();
+
+    useEffect(() => {
+        setIsLoading(true);
+        initSearchBar();
+        setIsLoading(false);
+    }, []);
+
+    /**
+     * Initialize the searchbar props with the corresponding index pattern and filters
+     */
+    const initSearchBar = async () => {
+        const indexPattern = await getIndexPattern(props?.defaultIndexPatternID);
+        setIndexPatternSelected(indexPattern);
+        const filters = await getInitialFilters(indexPattern);
+        filterManager.setFilters(filters);
+    }
+
+    /**
+     * Return the index pattern data by ID.
+     * If not receive a ID return the default index from the index pattern service
+     * @returns 
+     */
+    const getIndexPattern = async (indexPatternID?: string) => {
+        const indexPatternService = getDataPlugin().indexPatterns as IndexPatternsContract;
+        if(indexPatternID){
+            try{
+                return await indexPatternService.get(indexPatternID);
+            }catch(error){
+                // when the index pattern id not exists will get the default
+                console.error(error);
```

Comment:
It's ok to log the error or it's common behavior?

Reply:
It's a common behavior from osd indexPatternService but I want to leave a message to the developer. This case is when the hook receives an id but doesn't exist. This message can help to understand why the index pattern is not selected in the search bar